### DISCUSSION TITLE
What is expected in the Proposal?

### DISCUSSION THREAD

**Discussion by yoarora** [cncf/mentoring#913] [General]
Hello CNCF Maintainers and GSOC Mentors,
I wished to know what is required in a potential candidate's proposal for an idea for them to get selected? What level understanding of the idea, its potential fix, implementation, and timeline created is required? How long should the proposal be?

**Comment by nate-double-u** [comment]
Hi @arorasoham9, the [Google contributor's guide](https://google.github.io/gsocguides/student/) has some good information about proposals.

We've also been chatting about potential templates over in issue #622 which has some good resources listed.
